### PR TITLE
Speedup ingestion times

### PR DESCRIPTION
**Pull request by bendk**
Changed the ingestion code to prepare statements outside of any loops rather than creating them on each iteration.  Also, use tuples rather than named arguments.  This cuts ingestion time in half for the 2 record types that dominate the total time:

```
ingest/ingest-amp-wikipedia
                        time:   [446.26 ms 446.85 ms 447.53 ms]
                        change: [-51.727% -51.566% -51.408%] (p = 0.00 < 0.05)
                        Performance has improved.

ingest/ingest-amp-mobile
                        time:   [240.08 ms 240.25 ms 240.48 ms]
                        change: [-49.921% -49.801% -49.672%] (p = 0.00 < 0.05)
                        Performance has improved.
```

Refactored the prepare and execute calls into their own struct.  This way it's easy to make sure that the arguments line up with correctly with the statement placeholders.  I also think it simplifies things to remove the specifics of each SQL statement from the general ingestion logic.

### Pull Request checklist ###

- **Breaking changes**:  This PR follows our [breaking change policy](https://github.com/mozilla/application-services/blob/main/docs/howtos/breaking-changes.md)
  - [ ] This PR follows the breaking change policy:
     - This PR has no breaking API changes, or
     - There are corresponding PRs for our consumer applications that resolve the breaking changes and have been approved
- [ ] **Quality**: This PR builds and tests run cleanly
  - Note:
    - For changes that need extra cross-platform testing, consider adding `[ci full]` to the PR title.
    - If this pull request includes a breaking change, consider [cutting a new release](https://github.com/mozilla/application-services/blob/main/docs/howtos/cut-a-new-release.md) after merging.
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes a changelog entry in [CHANGELOG.md](../CHANGELOG.md) or an explanation of why it does not need one
  - Any breaking changes to Swift or Kotlin binding APIs are noted explicitly
- [ ] **Dependencies**: This PR follows our [dependency management guidelines](https://github.com/mozilla/application-services/blob/main/docs/dependency-management.md)
  - Any new dependencies are accompanied by a summary of the due dilligence applied in selecting them.

[Branch builds](https://github.com/mozilla/application-services/blob/main/docs/howtos/branch-builds.md): add `[firefox-android: branch-name]` to the PR title.

### CODE REVIEW

**Comment by bendk** [comment]
I saved the code I used to figure some of this stuff out in the [suggest-ingest-speedup-log](https://github.com/mozilla/application-services/compare/main...bendk:application-services:suggest-ingest-speedup-log) branch.  If we wanted to, we could also only go with speedup #1 in that branch, which would be a very simple fix.  However, I think the rest of the changes are good on their own anyways.

File: components/suggest/src/db.rs [hunk 1]
```diff
@@ -1407,6 +1210,133 @@ impl<'a> FullKeywordInserter<'a> {
...
+        keyword: &str,
+        full_keyword_id: Option<i64>,
+        rank: usize,
+    ) -> Result<()> {
+        self.0
+            .execute((suggestion_id, keyword, full_keyword_id, rank))?;
+        Ok(())
+    }
+}
+
```
Comment (bendk) [inline components/suggest/src/db.rs:1339]: We could define more of these statement types, for example to handle the inserting pocket/MDN/Weather data, or maybe even for the fetch/deletion code.  However, for this PR I only targetted the queries that I think would make a difference to the overall ingestion time for a fresh DB.

Comment (ncloudioj) [reply components/suggest/src/db.rs:1339]: Makes sense. Can you add a note for this to guide folks on when/how to adopt this treatment?

**Review by ncloudioj — APPROVED** [review]
Excellent! Thanks!

**Pull request merged**